build.gradle.kts

```diff
@@ -116,8 +116,6 @@ configureLinting(lintPaths)
 
 // Binary compatibility
 apiValidation {
-    nonPublicMarkers.add("aws.smithy.kotlin.runtime.InternalApi")
```

Comment:
In general I agree with this change but I am wondering about what we will do when we do need/want to break internal APIs. Presumably we'd deprecate the APIs and leave them in place for some period of time or perhaps indefinitely and switch codegen to use whatever the replacement is?

Reply:
Discussed offline and decided largely on what @aajtodd suggested: if we need to deprecate an `@InternalApi` declaration, we'll mark the API as `@Deprecated` and leave it in place for some number of releases. It will be considered for (potentially-breaking) removal if it is determined to be actively harming something (e.g., security, critical bugs, etc.).